nucypher/core.py

```diff
@@ -0,0 +1,259 @@
+        authorization = signer(kem_ciphertext_hash)
+
+        acp = AccessControlPolicy(
+            public_key=dkg_public_key,
+            conditions=conditions,
+            authorization=authorization,
+        )
+
+        # we need to link the ThresholdMessageKit to a specific version of the ACP
+        # because the ACP.aad() function should return the same value as the aad used
+        # for encryption. Since the ACP version can change independently of
+        # ThresholdMessageKit this check is good for code maintenance and ensuring
+        # compatibility - unless we find a better way to link TMK and ACP.
+        #
+        # TODO: perhaps this can be improved. You could have ACP be an inner class of TMK,
+        #  but not sure how that plays out with rust and python bindings... OR ...?
+        cls._validate_aad_compatibility(aad, acp.aad())
+
+        return ThresholdMessageKit(
+            kem_ciphertext,
+            dem_ciphertext,
+            acp,
+        )
+
+    def to_dict(self):
+        d = {
+            "version": self.version,
+            "kem_ciphertext": base64.b64encode(bytes(self.kem_ciphertext)).decode(),
+            "dem_ciphertext": base64.b64encode(self.dem_ciphertext).decode(),
+            "acp": self.acp.to_dict(),
+        }
+
+        return d
+
+    @classmethod
+    def from_dict(cls, message_kit: Dict) -> "ThresholdMessageKit":
+        return cls(
+            version=message_kit["version"],
+            kem_ciphertext=Ciphertext.from_bytes(
+                base64.b64decode(message_kit["kem_ciphertext"])
+            ),
+            dem_ciphertext=base64.b64decode(message_kit["dem_ciphertext"]),
+            acp=AccessControlPolicy.from_dict(message_kit["acp"]),
+        )
+
+    def __bytes__(self):
+        json_payload = json.dumps(self.to_dict()).encode()
+        b64_json_payload = base64.b64encode(json_payload)
+        return b64_json_payload
+
+    @classmethod
+    def from_bytes(cls, data: bytes) -> "ThresholdMessageKit":
+        json_payload = base64.b64decode(data).decode()
+        instance = cls.from_dict(json.loads(json_payload))
+        return instance
+
+
+class ThresholdDecryptionRequest(NamedTuple):
+    ritual_id: int
+    access_control_policy: AccessControlPolicy
+    variant: FerveoVariant
+    ciphertext: Ciphertext
+    context: Optional[Context]
+
+    def encrypt(
+        self, shared_secret: SessionSharedSecret, requester_public_key: SessionStaticKey
+    ) -> "EncryptedThresholdDecryptionRequest":
+        return EncryptedThresholdDecryptionRequest(
+            ritual_id=self.ritual_id,
+            requester_public_key=requester_public_key,
+            request_bytes=bytes(self),
+        )
+
+    def _to_dict(self):
+        d = {
+            "ritual_id": self.ritual_id,
+            "acp": self.access_control_policy.to_dict(),
+            "variant": "simple"
+            if self.variant == FerveoVariant.Simple
+            else "precomputed",
```

Comment:
It should be `str` serializable:
```
  assert str(FerveoVariant.Precomputed) == "FerveoVariant::Precomputed"
  assert str(FerveoVariant.Simple) == "FerveoVariant::Simple"
```

```suggestion
            "variant": str(self.variant)
```

Reply:
Yeah, but there is no from string conversion though, unless I'm mistaken...i.e. `FerveoVariant.from_str("FerveoVariant::Simple")`?